src/SmartComponents/AddSystemModal/AddSystemModal.js

```diff
@@ -128,10 +128,25 @@ export class AddSystemModal extends Component {
     }
 
     findNotInComparison(basketContent, comparedContent) {
+        let contentToRemove = [];
+
         if (comparedContent.length === 0) {
             return basketContent;
         } else {
-            return basketContent.filter(item => comparedContent.some(({ id }) => id !== item.id));
+            basketContent.forEach(function(basketItem) {
+                let exists = false;
+                comparedContent.forEach(function(comparedItem) {
+                    if (basketItem.id === comparedItem.id) {
+                        exists = true;
+                    }
+                });
+
+                if (!exists) {
+                    contentToRemove.push(basketItem);
+                }
+            });
+
+            return contentToRemove;
```

Comment:
Nit: This can be written functionally, just by using filter and find.

Reply:
Try smth like this. Hopefully it isn't flawed :slightly_smiling_face: 

```
basketContent.filter(basketItem => (
    comparedContent.findIndex(comparedItem => (basketItem.id === comparedItem.id)) === -1
));
```

Reply:
You fixed it!!! It worked like a charm. I'm still trying to learn how to combine these functions well.